std/haxe/ds/BalancedTree.hx

```diff
@@ -238,6 +431,15 @@ class BalancedTree<K, V> implements haxe.Constraints.IMap<K, V> {
 	}
 }
 
+/**
+	A key and its associated value.
+**/
+typedef Entry<K,V> = {key:Null<K>, value:Null<V>};
```

Comment:
I didn't read this PR's manual yet but I guess these types should be private.

Reply:
Possibly, although there are use-cases where someone might want to hang onto one of the returned entry structures for longer than just an immediate member access.

Reply:
You can still do that, you just cannot reference the type explicitly from the outside. The problem with public types like this is that they become `haxe.ds.Entry` in the output because Haxe's modules are not represented at run-time.

Reply:
Makes sense, I just put them there with the same access as the existing `TreeNode`. Perhaps all of these types should be made private, actually.